pkg/logentry/stages/drop.go

```diff
@@ -0,0 +1,190 @@
+package stages
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/loki/pkg/logql"
+)
+
+package stages
+
+import (
+"time"
+
+"github.com/prometheus/prometheus/pkg/labels"
+
+"github.com/go-kit/kit/log"
+"github.com/mitchellh/mapstructure"
+"github.com/pkg/errors"
+"github.com/prometheus/client_golang/prometheus"
+"github.com/prometheus/common/model"
+
+"github.com/grafana/loki/pkg/logql"
+)
+
+const (
+	ErrEmptyMatchStageConfig = "match stage config cannot be empty"
+	ErrPipelineNameRequired  = "match stage pipeline name can be omitted but cannot be an empty string"
+	ErrSelectorRequired      = "selector statement required for match stage"
+	ErrMatchRequiresStages   = "match stage requires at least one additional stage to be defined in '- stages'"
+	ErrSelectorSyntax        = "invalid selector syntax for match stage"
+	ErrStagesWithDropLine    = "match stage configured to drop entries cannot contains stages"
+	ErrUnknownMatchAction    = "match stage action should be 'keep' or 'drop'"
+	MatchActionKeep          = "keep"
+	MatchActionDrop          = "drop"
+)
+
+// DropConfig contains the configuration for a dropStage
+type DropConfig struct {
+	Source     *string `mapstructure:"source"`
+	Value      *string `mapstructure:"value"`
+	Expression *string `mapstructure:"expression"`
+	regex      *regexp.Regexp
+	OlderThan  *string `mapstructure:"older_than"`
+	olderThan  time.Duration
+	LongerThan *string `mapstructure:"longer_than"`
```

Comment:
can we use https://github.com/grafana/loki/blob/master/pkg/util/flagext/bytesize.go#L11